doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,199 @@
+when you need to update the configuration process.
+
+## Makefile
+
+This file is not generated, it is maintained by humans. Note that this is
+not usually run on Windows, where [vcbuild.bat](#vcbuild.bat) is used instead.
+
+### Options
+
+- `-j <n>`: number of threads used to build the binary. Note that on the non-ci
+  targets, the parallel tests will take up all the available cores, regardless
+  of this option.
+
+### Targets
+
+For the targets listed below that do not have detailed descriptions
+in this document, please see their comments in the Makefile instead.
+
+Here is a list of targets that will be commonly triggered or updated
+by its users:
+
+#### General Targets
+
+Consider updating these targets when you update the build process:
+
+- `all`: the entry point, called when running `make` without specifying a
+  target.
+- `clean`
+- `install`
+- `uninstall`
+
+#### Test Targets for Developers
+
+Consider updating these targets when you update the testing process for
+developers:
+
+- `test`: this is meant to be run by the developers.
+- `test-only`: this does not run linter or build the documentation.
+- `coverage`
+- `coverage-clean`
+
+#### CI Targets
+
+Consider updating these targets when you update the testing process on CI.
+Note that the CI runs different targets on Raspberry Pis
+(`node-test-commit-arm-fanned`) and Linux with FIPS
+(`node-test-commit-linux-fips`).
+
+To update the process for most CI, update:
+
+- `run-ci`: does not affect the CI job on Raspberry Pis and Linux with FIPS.
+- `build-ci`: does not affect the CI job on Linux with FIPS.
+- `test-ci`: does not affect the CI job on Raspberry Pis
+
+To update the `node-test-linter` CI job, update:
+
+- `lint-ci`
+- `lint-js-ci`
+
+For the Raspberry Pis, update:
+
+- `test-ci-js`
+- `test-ci-native`
+
+#### `deps` Testing Targets 
+
+The normal testing process does not run tests of third-party libraries
+inside `deps`, because most of them can run their tests on their own. However,
+there are a few targets in the Makefile that are used to test V8 and npm,
+because we have our own floating patches (not merged in the upstream) that
+need to be tested. Consider updating these targets when you need to update
+the process of testing them:
+
+For `deps/v8`:
+
+- `v8`: rebuilds `deps/v8` as a git tree, pulls the third-party dependencies
+  and builds V8
+- `test-v8`: run by the `node-test-commit-v8-linux` CI job
+- `test-v8-benchmarks`
+- `test-v8-intl`
+- `test-v8-all`
+
+For `deps/npm`:
+
+- `test-npm`
+- `test-npm-publish`
+
+#### Other Test Targets
+
+The following targets use `node-gyp` bundled inside the npm in the codebase:
+
+- `test-addons`
+- `test-addons-napi`
+- `test-gc`
+
+There are many `test-*` targets that can be run to test a specific part of
+the codebase, most of them using `tools/test.py`. There is also:
+
+- `cctest`: run the C++ tests.
+
+#### Documentation Targets
+
+The Node.js API documentation is built within this codebase. Consider updating
```

Comment:
IMO, it's not entirely clear what 'within this codebase' means.  Maybe write this as "... built from the Markdown files in `doc/`."